### PR TITLE
Add safeTxGas to Safe transactions

### PR DESCRIPTION
**Pull request by germartinez**
Add safeTxGas to Safe transactions

### CODE REVIEW

**Comment by rmeissner** [comment]
Would need to look into it in detail, but I would guess the error you see is because you use ganache without the appropriate flag ;)

But yeah you probably need to use the low level eth_call to do this (but this was kind of expected)

File: src/utils/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    const estimateResponse = await cpkProvider.checkSingleCall({
+      from: safeAddress,
+      to: safeAddress,
+      data: estimateData,
+    });
+
+    const txGasEstimate = new BigNumber(estimateResponse.substring(138), 16);
+
+    // Add 10k else we will fail in case of nested calls
+    const safeTxGas = txGasEstimate.toNumber() + 10000;
```
Comment (rmeissner) [inline src/utils/index.js:31]: This will not work for all cases, there is an open pr on the safe-co tracts repo for this

Comment (rmeissner) [reply src/utils/index.js:31]: https://github.com/gnosis/safe-contracts/pull/188/files#diff-0003dd2226a508305de12e134ec0fe66R60

**Comment by germartinez** [comment]
Tests only run when executing them like this:

One console:
`ganache-cli -d --noVMErrorsOnRPCResponse`

Other console (root folder of the project):
`yarn test-norpc --network local`

Otherwise, they fail

File: src/utils/index.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
+const { BigNumber } = require('bignumber.js');
+const safeAbi = require('../abis/SafeAbi.json');
+const { zeroAddress } = require('./constants');
+
+const baseGasValue = (hexValue) => {
+  switch (hexValue) {
+  case '0x': return 0;
+  case '00': return 4;
+  default: return 68;
```
Comment (rmeissner) [inline src/utils/index.js:9]: after the last hardfork it is not 68 anymore but 16

File: src/utils/index.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  operation,
+  safeTxGas,
+  gasToken,
+  refundReceiver,
+  signatureCount,
+  nonce,
+  gasPrice,
+) => {
+  // (array count (3 -> r, s, v) + ecrecover costs) * signature count
+  const signatureCost = signatureCount * (68 + 2176 + 2176 + 6000);
```
Comment (rmeissner) [inline src/utils/index.js:33]: this is actually not important here, check https://github.com/gnosis/safe-react/pull/752/files

**Pull request closed without merging**